static/js/message_events.js

```diff
@@ -176,17 +177,20 @@ export function update_messages(events) {
 
         const new_stream_id = event.new_stream_id;
 
+        const old_stream = sub_store.get(event.stream_id);
+
         // A topic edit may affect multiple messages, listed in
         // event.message_ids. event.message_id is still the first message
         // where the user initiated the edit.
         topic_edited = new_topic !== undefined;
         stream_changed = new_stream_id !== undefined;
-        if (topic_edited || stream_changed) {
+        stream_not_archived = old_stream !== undefined;
+        if ((topic_edited || stream_changed) && stream_not_archived) {
             const going_forward_change = ["change_later", "change_all"].includes(
                 event.propagate_mode,
             );
 
-            const stream_name = sub_store.get(event.stream_id).name;
+            const stream_name = old_stream.name;
```

Comment:
Given the `stream_name` in the check within this block below, I think we want to arrange things so that `stream_name` is undefined in this case, rather than this block not being run.